xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-api/src/main/java/org/xwiki/panels/internal/AbstractPanelsUIExtensionManager.java

```diff
@@ -95,24 +99,30 @@
 
         // Verify that there's a panel configuration property defined, and if not don't return any panel extension.
         if (!StringUtils.isEmpty(panelConfigurationString)) {
-            List<String> panelSerializedReferences = new ArrayList<String>();
+            Set<DocumentReference> panelSerializedReferences = new HashSet<>();
             for (String serializedReference : getConfiguration().split(",")) {
-                panelSerializedReferences.add(serializer.serialize(resolver.resolve(serializedReference.trim())));
+                panelSerializedReferences.add(resolver.resolve(serializedReference.trim()));
             }
 
             try {
                 List<UIExtension> allExtensions =
                     contextComponentManagerProvider.get().getInstanceList(UIExtension.class);
-                for (String panelSerializedReference : panelSerializedReferences) {
-                    // TODO: This is not performant and will not scale well when the number of UIExtension instances
-                    // increase in the wiki
-                    for (UIExtension extension : allExtensions) {
-                        // Resolve and serialize the extension id to make it absolute before we can compare it with
-                        // the serialized panel reference.
-                        String extensionId = serializer.serialize(resolver.resolve(extension.getId()));
-                        if (extensionId.equals(panelSerializedReference)) {
-                            panels.add(extension);
-                        }
+                // TODO: This is not performant and will not scale well when the number of UIExtension instances
+                // increase in the wiki
+                for (UIExtension extension : allExtensions) {
+                    DocumentReference extensionId;
+
+                    // in this case we can directly retrieve the document reference which gives also the wiki name
+                    // allows to fix XWIKI-14557
```

Comment:
it's XWIKI-15601. 

However, in general we try to not reference jira issue too much as we consider jira to be transient. So instead we explain the problem we're fixing.

Reply:
It seems that you improved performance by reusing the doc ref that is already stored in the WikiComponent instance.

Reply:
Now that doesn't explain what was the problem with XWIKI-15601

Reply:
You need to check that https://jira.xwiki.org/browse/XWIKI-15388 is still working